src/component/completed_list.rs

```diff
@@ -43,6 +44,24 @@ impl CompletedList {
             "Restores the selected task",
         )]
     }
+
+    pub fn restore_task(&mut self, app: &mut App) {
+        if app.task_store.completed_tasks.is_empty() {
+            return;
+        }
+        app.task_store.tasks.push(Task::from_completed_task(
+            app.task_store.completed_tasks.remove(*self.selected()),
+        ));
+        if *self.selected() == app.task_store.completed_tasks.len()
+            && !app.task_store.completed_tasks.is_empty()
+        {
+            *self.selected_mut() -= 1;
+        }
```

Comment:
Also, could you explain to me what is the purpose of these lines?

Reply:
If the selected task that was restored is at the bottom, it would be removed from the list and cause an index out-of-bounds scenario, this prevents that.

Reply:
Oh, sure, I was just thinking about the normal `TaskList` and that confused me. Thanks